bullet_train-super_load_and_authorize_resource/test/controllers/concerns/bullet_train/loads_and_authorizes_resource_test.rb

```diff
@@ -1,23 +1,87 @@
 require "test_helper"
 
 class BulletTrain::LoadsAndAuthorizeResourceTest < ActiveSupport::TestCase
-  class TestClass
+  class TestControllerClass < ActionController::Base
     include BulletTrain::LoadsAndAuthorizesResource
+
+    attr_accessor :child_object, :current_user, :parent_object, :team
+
+    def self.regex_to_remove_controller_namespace
+    end
+
+    def can?(*args)
+    end
+  end
+
+  module Users
+    class TestControllerClass < ActionController::Base
+      include BulletTrain::LoadsAndAuthorizesResource
+
+      def self.regex_to_remove_controller_namespace
+        /^BulletTrain::LoadsAndAuthorizeResourceTest::/
+      end
+    end
   end
 
-  test "it defines .model_namespace_from_controller_namespace" do
-    assert TestClass.respond_to?(:model_namespace_from_controller_namespace)
+  test "model_namespace_from_controller_namespace returns an array of modules names based on the classes namespace minus regex_to_remove_controller_namespace" do
+    assert_equal ["BulletTrain", "LoadsAndAuthorizeResourceTest"], TestControllerClass.model_namespace_from_controller_namespace
+    assert_equal ["Users"], Users::TestControllerClass.model_namespace_from_controller_namespace
   end
 
   test "it defines .account_load_and_authorize_resource" do
-    assert TestClass.respond_to?(:account_load_and_authorize_resource)
+    assert TestControllerClass.respond_to?(:account_load_and_authorize_resource)
+  end
+
+  test "it defines .regex_to_remove_controller_namespace" do
+    assert TestControllerClass.respond_to?(:regex_to_remove_controller_namespace)
   end
 
   test "it defines #load_team" do
-    assert TestClass.new.respond_to?(:load_team)
+    assert TestControllerClass.new.respond_to?(:load_team)
+  end
+
+  test "#load_team returns nil child_object and parent_object are nil" do
+    subject = TestControllerClass.new
+    subject.child_object = nil
+    subject.parent_object = nil
+
+    assert_nil subject.load_team
+  end
+
+  test "#load_team sets Current attributes if defined" do
+    current_class = Class.new(ActiveSupport::CurrentAttributes) do
+      attribute :team
+    end
+    Object.const_set(:Current, current_class)
```

Comment:
Will this override an actual `Current` constant in apps? I wonder if this is related to why starter repo tests fail.

I wonder if we'd be better off with something like this in `test_helper.rb`:

```ruby
# Define a fallback `Current` when we're running our tests isolated from the starter repo's `Current`.
unless defined?(::Current)
  class Current < ActiveSupport::CurrentAttributes
    attribute :team
  end
end
```

Reply:
This may be one of those things where writing tests directly in the gem is somewhat lacking compared to writing tests in the starter repo where we have a more robust BT implementation that we can rely on.

Reply:
I've updated the test as @kaspth suggested.

I think having to write tests in another application or library because they won't work in the library where the code under-test is written is an indication of one or more software design problems.

`bullet_train-super_load_and_authorize_resource` has some dependency issues.

Reply:
For example, the code of this gem has an undeclared dependency on `cancancan` or, at a minimum, that the class calling `account_load_and_authorize_resource` defines a method `load_and_authorize_resource`.

Reply:
Yeah there are definitely some undeclared dependencies and circular dependencies through a few of the BT packages. I have it on my list to try to sort that out soon.